pombola/south_africa/data/members-interests/convert_to_import_json.py

```diff
@@ -417,25 +537,32 @@ def mp_to_person_slug(self, mp):
             return person.slug
         except Person.DoesNotExist:
             try:
-                name_parts = re.findall(r'(.*?), (.*)', muddled_name)
-                person = Person.objects.get(Q(slug__contains=slugify(name_parts[0][0])) & Q(slug__contains=slugify(name_parts[0][1])))
-                return person.slug
+                name_base = re.findall(r'(.*?), (.*)', muddled_name.replace('-', ','))
+                if name_base:
+                    name_parts = name_base[0]
+                    try:
+                        person = Person.objects.get(Q(slug__contains=slugify(name_parts[0])) & Q(slug__contains=slugify(name_parts[1])))
+                        return person.slug
+                    except Person.MultipleObjectsReturned:
+                        print('Multiple people matched', name_parts, slug if slug else '')
             except Person.DoesNotExist:
                 last_name = name.split(' ')[-1]
 
-                possible_persons = Person.objects.filter(legal_name__icontains=last_name)
+                possible_persons = Person.objects.filter(
+                    legal_name__icontains=last_name)
 
                 if self.finding_slug_corrections and possible_persons.count() == 1:
                     possible_slug = possible_persons.all()[0].slug
                     self.slug_corrections[slug] = possible_slug
                     return possible_slug
 
                 for person in possible_persons:
-                    print 'perhaps: "{0}": "{1}",'.format(slug, person.slug)
+                    print('perhaps: "{0}": "{1}",'.format(slug, person.slug))
                 else:
-                    print "no possible matches for {0}".format(slug)
+                    print("no possible matches for {0}".format(slug))
 
-                raise Exception("Slug {0} not found, please find matching slug and add it to the slug_corrections".format(slug))
+                raise Exception(
+                    "Slug {0} not found, please find matching slug and add it to the slug_corrections".format(slug))
```

Comment:
if this printed rather than raised, we could run through once and get all the ones that don't match, right? as it stands, you have to run a massive job over and over, adding slug mappings one by one?